api/src/main/java/org/apache/iceberg/expressions/Predicate.java

```diff
@@ -19,15 +19,49 @@
 
 package org.apache.iceberg.expressions;
 
+import com.google.common.base.Preconditions;
+import java.util.Set;
+
 public abstract class Predicate<T, R extends Reference> implements Expression {
   private final Operation op;
   private final R ref;
   private final Literal<T> literal;
+  private final LiteralSet<T> literalSet;
```

Comment:
Why are these stored in the superclass? It is strange to have both `literal` and `literalSet` in all implementations. Can these be fields in the concrete implementations?

Reply:
I think the only place where these are used is in `toString`, right?

Reply:
Thanks for the comments. Updated the classes accordingly.